performance_manager/lib/postgres_utils.py

```diff
@@ -76,22 +79,114 @@ def get_aws_engine() -> sa.future.engine.Engine:  # type: ignore
 # https://github.com/python/mypy/issues/2477
 
 
-def write_from_dict(
-    input_dictionary: dict,
-    sql_session: sessionmaker,
-    destination_table: sa.Table,
-) -> None:
+class DatabaseManager:
     """
-    try to write a dict to a table. if experimental, use the testing engine,
-    else use the local one.
+    manager class for rds application operations
     """
-    try:
-        with sql_session.begin() as session:  # type: ignore
-            session.execute(sa.insert(destination_table), input_dictionary)
-            session.commit()
-    except Exception as e:
-        logging.error("Error Writing Dataframe to Database")
-        logging.exception(e)
+
+    def __init__(self, experimental: bool = False, verbose: bool = False):
+        """
+        initialize db manager object, creates engine and sessionmaker
+        """
+        if experimental:
+            self.engine = get_experimental_engine(echo=verbose)
+
+            @sa.event.listens_for(sa.engine.Engine, "connect")
+            def set_sqlite_pragma(dbapi_connection, _):  # type: ignore
+                cursor = dbapi_connection.cursor()
+                cursor.execute("PRAGMA foreign_keys=ON")
+                cursor.close()
```

Comment:
what does this do?

Reply:
per this: https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#foreign-key-support

it is the way to add foreign key support of sqlite, will detail in a comment